### ISSUE TITLE
Gcop 537

### ISSUE THREAD

**Issue by Karvan** [Geeksltd/GCop#10]
Gcop 537 is showing false alarm.
 
```
 var data = ControllerContext.RequestContext.RouteData;
 if (...  && data.Values["clientUrlExtension"] == null)
```
Warning 537: File extensions contain the leading dot character.

**Comment by Foroughi** [comment]
We need to be more specific on this rule. I believe this rule should kick in only if : 

1 - we have `Equals Expression` (== operator)
2 - one side is a `Literal Expression Syntax` (a hard-coded string)
3 - the other side is `System.IO.FileInfo.Extension`

ain't that true ?!

**Comment by Foroughi** [comment]
Another possible scenario is this : 

```
string str = new System.IO.FileInfo("file address").Extension;
if(str == ".txt")
{
       //bla bla 
}
```


which without a code normalizer to convert that piece of code the following sample, is not possible 

```
if(new System.IO.FileInfo("file address").Extension== ".txt")
{
       //bla bla 
}
```

**Comment by Foroughi** [comment]
@reporter : Fixed on SlnMerge/03b16b3d10beaa19bff2afee8061678343ec27e5
